portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -24,9 +27,27 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
 
+  private ArrayList<String> comments;
```

Comment:
Prefer using interfaces instead of concrete types. `ArrayList` is a concrete implementation, so in this example, I would recommend using `List<String>`. 

Moreover, you can choose to initialize it directly:

```java
private List<String> comments = List.of(
    "comment1",
    "comment2",
    "..."
);
```

`List.of` returns an unmodifiable list of elements, which  is preferred. The general rule is: "always prefer immutable data structures whenever possible."